compiler/include/symbol.h

```diff
@@ -361,6 +379,45 @@ class ArgSymbol : public LcnSymbol {
 *                                                                   *
 ********************************* | ********************************/
 
+class ShadowVarSymbol : public VarSymbol {
+public:
+  ShadowVarSymbol(ForallIntentTag iIntent, const char* iName, Expr* iSpec = NULL);
+
+  virtual void    verify();
+  virtual void    accept(AstVisitor* visitor);
+  DECLARE_SYMBOL_COPY(ShadowVarSymbol);
+
+  virtual void    replaceChild(BaseAST* oldAst, BaseAST* newAst);
+  virtual bool    isConstant()                              const;
+  virtual bool    isConstValWillNotChange()                 const;
+
+  const char* intentDescrString() const;
+  // The corresponding outer var or NULL if not applicable.
+  SymExpr* outerVarSE() const { return (SymExpr*)outerVarRep; }
+  bool     isReduce()   const { return intent == TFI_REDUCE;  }
+  Expr*    spec()       const;
+  void     removeSupportingReferences();
+
+  ForallIntentTag intent;
+
+  // This would be a SymExpr*, if not for checkIdInsideWithClause().
+  // See also: sv->outerVarSE() and outerVarSym(sv).
+  Expr* outerVarRep;
+
+  // For a reduce intent, the reduce expression.
+  // For a task-private variable, the initialization expression.
+  // Either way, wrapped in a block.  Otherwise it is NULL.
+  BlockStmt* specBlock;
+
+  Symbol* reduceGVar;
+  bool pruneit;
```

Comment:
I think it's debatable whether or not we should have any public fields in a class like this. But pruneit seems "obviously" private to me. Is that impossible to do in an incremental way in this implementation? Would it be easier to remove pruneit entirely, say?

Reply:
While it is indeed obviously private, it is both read and written in implementForallIntents.cpp in non-method functions. So marking it as "private" will prevent that code from compiling.

My hopes are to eliminate "pruning" altogether so that this field can be removed. OK to leave it as-is for now?

Reply:
Yes but let's at least put a comment about how we plan to get rid of it.